plugins/datalogger/src/datamonitor/datamonitormodel.cpp

```diff
@@ -200,11 +222,40 @@ void DataMonitorModel::addValue(double time, double value)
 		setMaxValue(value);
 	}
 
+	// Apply scaling, offset, and unit of measurement scale
+	double adjustedValue = (value + m_offset) * m_scale;
```

Comment:
Will the overriden offset be provided in counts as well? If the new offset is provided as UM (V, Celsius, etc) it should be applied after scaling, otherwise it should be applied before before.